common/src/main/java/org/conscrypt/ConscryptEngine.java

```diff
@@ -1681,7 +1681,11 @@ private void closeAndFreeResources() {
     @SuppressWarnings("deprecation")
     protected void finalize() throws Throwable {
         try {
-            closeAndFreeResources();
+            if (ssl != null) {
```

Comment:
nit: would be nice to add a one-line comment why it is needed (optional as CL description is good)

Reply:
Good point, done.